src/iterative_ensemble_smoother/experimental.py

```diff
@@ -276,30 +239,21 @@ def correlation_threshold(ensemble_size: int) -> float:
             assert cov_YY.ndim == 2, "'cov_YY' must be a 2D array"
             assert cov_YY.shape == (Y.shape[0], Y.shape[0])
 
+        # Identify rows with at least one significant correlation.
+        significant_rows = np.any(significant_corr_XY, axis=1)
+
         # TODO: memory could be saved by overwriting the input X
         X_out: npt.NDArray[np.double] = np.copy(X)
-        for (unique_row, indices_of_row) in groupby_indices(significant_corr_XY):
-
-            if verbose:
-                print(
-                    f"    Assimilating {len(indices_of_row)} parameters"
-                    + " with identical correlation thresholds to responses."
-                )
-                print(
-                    "    The parameters are significant wrt "
-                    + f"{np.sum(unique_row)} / {len(unique_row)} responses."
-                )
-
-            # These parameters are not significantly correlated to any responses
-            if np.all(~unique_row):
-                continue
+        # Loop only over rows with significant correlations
+        for indices_of_row in np.where(significant_rows)[0]:
+            unique_row = significant_corr_XY[indices_of_row]
 
             # Get the parameters (X) that have identical significant responses (Y)
             X_subset = X[indices_of_row, :]
```

Comment:
Since `indices_of_row` is now an int, this variable `X_subset` is now a 1D array. I am surprised that an error is not raised or that the code does not fail?

Perhaps look at this quickly before merging. Also, perhaps rename 
```
indices_of_row -> parameter_num
unique_row -> correlated_responses
```
or something like that. No other comments.

Reply:
Have not looked into it deeply but I'm assuming broadcasting is helping us out in this: `X_subset + cov_XY_subset @ T`.
Thanks for the better variable names 👍